### PR TITLE
Add overseer collection for page views and events

### PR DESCRIPTION
**Pull request by jnordberg**
Haven't ripped out the old tracking code yet since we need it for pageview counts

steem-js changes https://github.com/steemit/steem-js/compare/5f50ad5...ec4439c1319e21ccb4b52a3de3798a37b84b0001

### CODE REVIEW

**Comment by valzav** [comment]
What is overseer and why are we integrating it?

**Comment by jnordberg** [comment]
[Overseer](https://github.com/steemit/overseer) is the new analytics engine

**Comment by valzav** [comment]
@jnordberg, is it already supported by backend? If it is I can merge this pull request, if not please mark it as wip.

**Comment by jnordberg** [comment]
It is already supported but only in dev, blocked by jussi getting in prod so I'll mark it as WIP

**Comment by sneak** [comment]
This should cause no problems being deployed in prod even if jussi/overseer is down - it should fail gracefully. That said, if it doesn’t it’s a bug that needs fixing, if it does then merging it in and deploying it to prod now won’t hurt anything.

**Comment by sneak** [comment]
I want to further review the steem-js change here - please don’t merge this yet.

File: src/app/utils/ServerApiClient.js [hunk 1]
```diff
@@ -51,12 +55,11 @@ export function markNotificationRead(account, fields) {
 }
 
 let last_page, last_views, last_page_promise;
-export function recordPageView(page, ref) {
+export function recordPageView(page, ref, account) {
     if (last_page_promise && page === last_page) return last_page_promise;
-    if (window.ga) { // virtual pageview
```
Comment (sneak) [inline src/app/utils/ServerApiClient.js]: please don't neuter GA here - this code should be removed only ifwhen GA is removed entirely.

please try to make PRs single-issue.  if you want to remove GA at the same time, make a second PR that does just that.  by adding overseer collection and removing GA, it makes the title of the PR misleading.

Comment (jnordberg) [reply src/app/utils/ServerApiClient.js]: Ok, I'll put it back

**Comment by relativityboy** [comment]
@jnordberg @valzav - I think it's time to move to feature flags. This would be a nice, simple place to start.

**Comment by jnordberg** [comment]
@relativityboy  I want this in prod as soon as we have jussi there to receive the events so I can start working with the data. Rather not have it hang on another feature being implemented

**Comment by jnordberg** [comment]
@sneak we good on the steem-js changes?

File: src/app/utils/ServerApiClient.js [hunk 2]
```diff
@@ -51,12 +55,15 @@ export function markNotificationRead(account, fields) {
...
 let last_page, last_views, last_page_promise;
-export function recordPageView(page, ref) {
+export function recordPageView(page, ref, account) {
     if (last_page_promise && page === last_page) return last_page_promise;
     if (window.ga) { // virtual pageview
         window.ga('set', 'page', page);
         window.ga('send', 'pageview');
     }
+    api.call('overseer.pageview', {page, referer: ref, account}, (error) => {
+        if (error) console.warn('overseer error', error, error.data)
```
Comment (goldibex) [inline src/app/utils/ServerApiClient.js]: missing semicolons here. the surrounding code uses them, please add

File: src/app/utils/ServerApiClient.js [hunk 3]
```diff
@@ -31,6 +32,9 @@ export function serverApiRecordEvent(type, val, rate_limit_ms = 5000) {
     const value = val && val.stack ? `${val.toString()} | ${val.stack}` : val;
     const request = Object.assign({}, request_base, {body: JSON.stringify({csrf: $STM_csrf, type, value})});
     fetch('/api/v1/record_event', request);
+    api.call('overseer.collect', {collection: 'event', metadata: {type, value}}, (error) => {
+        if (error) console.warn('overseer error', error, error.data)
```
Comment (goldibex) [inline src/app/utils/ServerApiClient.js]: missing semicolons here. the surrounding code uses them, please add

**Pull request merged**